### PR TITLE
1. Create profiles to store additional data for Candidate vs. Employe…

### PR DESCRIPTION
**Pull request by daisycrego**
…r with a OneToOne reference to CustomUser, the underlying user model still being used for authentication.

2. Separate out register into candidate_register and employer_register. Each view still creates the CustomUser object. The employer_register view creates an additional Department object to be stored in the Employer profile along with the CustomUser object. The candidate_register view creates a CandidateProfile object which is stored alongside the CustomUser object in the Candidate profile.
3. Add booleans (is_candidate, is_employer) to the CustomUser to prevent cross-login between Employers and Candidates (e.g. Candidate cannot login through Employer login page, even though both are CustomUsers).
4. Override the confirm_login_allowed() method of the EmployerLoginForm and CandidateLoginForm to check the is_candidate and is_employer booleans before logging the user in, generating a form ValidationError otherwise.
5. Create decorators similar to the @login_required decorator, except checking that @candidate_login_required and @employer_login_required
6. Override eq and ne for CandidateProfile, Candidate, and Employer.

### CODE REVIEW

**Comment by jackxujh** [comment]
Looks this pr is jam-packed. Just one little thing, maybe try to use titles (a short yet descriptive first line of the commit message) in future commits and pull requests. Preferably in less than 50 characters. Having ellipsis in the title reduces readability. If the title does not fit, you may consider rephrasing or breaking down the commit.

**Comment by daisycrego** [comment]
> Looks this pr is jam-packed. Just one little thing, maybe try to use titles (a short yet descriptive first line of the commit message) in future commits and pull requests. Preferably in less than 50 characters. Having ellipsis in the title reduces readability. If the title does not fit, you may consider rephrasing or breaking down the commit.

Sorry about that @jackxujh. i definitely agree, in the future I will try to break the actions down into commits rather than doing 1 commit at the end. It was broken up right until the end so I didn't want to commit anything :) but you're right

File: register/views.py [hunk 1]
```diff
@@ -15,12 +16,46 @@ def register(request):
...
+        return HttpResponseRedirect(reverse("uplyft:index"))
+    if request.method == "POST":
+        form = EmployerRegistrationForm(request.POST)
+        if form.is_valid():
+            user = form.save(commit=True)
+            email = form.cleaned_data.get("email")
+            password = form.cleaned_data.get("password1")
+            department = form.cleaned_data.get("department")
+            user.is_employer = True
+            user.is_candidate = False
```
Comment (surajgovardhangaikwad) [inline register/views.py]: I'm just thinking if we need 2 boolean fields to check the user type. 
Wouldn't it probably be easier if we just had one boolean field for it? Or is there a reason why we're storing 2 boolean fields is_employer and is_candidate?

**Comment by surajgovardhangaikwad** [comment]
> 
> 
> > Looks this pr is jam-packed. Just one little thing, maybe try to use titles (a short yet descriptive first line of the commit message) in future commits and pull requests. Preferably in less than 50 characters. Having ellipsis in the title reduces readability. If the title does not fit, you may consider rephrasing or breaking down the commit.
> 
> Sorry about that @jackxujh. i definitely agree, in the future I will try to break the actions down into commits rather than doing 1 commit at the end. It was broken up right until the end so I didn't want to commit anything :) but you're right

Yeah, that happens sometimes, usually what I do is that I copy my entire commit message in the description and maybe just put a short sentence describing the big picture of the changes

**Pull request merged**